internal/core/data/v2/application/event.go

```diff
@@ -26,14 +27,14 @@ import (
 
 // The AddEvent function accepts the new event model from the controller functions
 // and invokes addEvent function in the infrastructure layer
-func AddEvent(e models.Event, ctx context.Context, dic *di.Container) (string, error) {
+func AddEvent(e models.Event, ctx context.Context, dic *di.Container) (id string, err v2.EdgexError) {
```

Comment:
Here you return `v2.EdgexError`, but in other places you left the return type as `error`. Any reason here you use `v2.EdgexError`?